src/Tokenizer/Tokens.php

```diff
@@ -150,22 +150,19 @@ public static function detectBlockType(Token $token): ?array
      * Create token collection from array.
      *
      * @param array<int, Token> $array       the array to import
-     * @param ?bool             $saveIndices save the numeric indices used in the original array, default is yes
+     * @param bool              $saveIndices
      */
-    public static function fromArray($array, $saveIndices = null): self
+    public static function fromArray($array, $saveIndices = false): self
     {
-        $tokens = new self(\count($array));
+        if (!array_is_list($array) && false !== $saveIndices) {
+            throw new \InvalidArgumentException('Tokens must be a list - saving indices is no longer supported');
```

Comment:
BC break

Reply:
In theory, described in the PR description why it is there, in practice, noone is using it.

Reply:
assumption

Reply:
yes

Reply:
BC is not acceptable.
If you want to, trigger deprecation with `Utils::triggerDeprecation` and don't change the behaviour.

Reply:
It is edge case allowed in past in theory, but as the methods basically all iterate using increment/decrement, I do not think it was ever fully supported. As stated in the description, I belive it is a consequence of `SplFixedArray::fromArray()` method signature. A minor version release should be fine with this PR as is.

Reply:
note for myself how to do it: #7800

Reply:
done